src/index.js

```diff
@@ -371,6 +373,40 @@ class Ninja {
     })
   }
 
+  /**
+   * Verifies the incoming transaction
+   * @param {Object} obj
+   * @param {string} obj.senderPaymail
+   * @param {string} obj.senderIdentityKey // OR ?
+   * @param {string} obj.referenceNumber
+   * @param {string} obj.description
+   * @returns {Promise<Boolean>} A success boolean status
+   */
+  async verifyIncomingTransaction ({ senderPaymail, senderIdentityKey, referenceNumber, description, amount }) {
+    // Check if the referenced transaction is pending
+    const [transaction] = await this.getPendingTransactions({ referenceNumber })
+    if (transaction && transaction.status === 'unprocessed') {
+      const [outputs] = transaction.outputs
```

Comment:
```suggestion
      const [firstOutput] = transaction.outputs
```
This destructures the array, and checks the first output only. We should check every output, because the Paymail BRFC spec says that multiple oututs could be provided.

Is the amount of the transaction returned? If so, we should check the amount of the transaction, rather than to check the first output only.

Allowing multiple outputs lets people split things up for privacy, since no one knows if it's all going to the same person, change, fees of some kind, etc. For example, the `paymail` package (when it sent the payment), could have divided the amount in half, and split the payment up across two outputs (each with a separate `derivationSuffix`, but on the same `derivationPrefix`).

Reply:
See [latest commit](https://github.com/p2ppsr/utxoninja/pull/11/commits/b142ae33c3c364e677f4acb75b136b95d64e5cf9) which validates the senderIdentityKey for every output.